### PR TITLE
security center: fix acctest

### PR DESCRIPTION
**Pull request by ziyeqf**
the create/update operation appears to be eventually consistent, which caused the acctests flakey.

Screenshot 2023-08-09 at 14 43 24

### CODE REVIEW

File: internal/services/securitycenter/advanced_threat_protection_resource.go [hunk 1]
```diff
@@ -88,6 +88,34 @@ func resourceAdvancedThreatProtectionCreateUpdate(d *pluginsdk.ResourceData, met
...
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending: []string{"diff"},
+		Target:  []string{"consistent"},
+		Refresh: func() (result interface{}, state string, err error) {
+			resp, err := client.Get(ctx, id.TargetResourceID)
+			if err != nil {
+				return resp, "error", err
+			}
+			if atpp := resp.AdvancedThreatProtectionProperties; atpp != nil {
+				if atpp.IsEnabled != nil && *atpp.IsEnabled == d.Get("enabled").(bool) {
```
Comment (jackofallops) [inline internal/services/securitycenter/advanced_threat_protection_resource.go:103]: `nil` and `false` here could be the same thing? Are we certain that in the not enabled case we always get a value back here?

**Review by jackofallops — COMMENTED** [review]
Hi @ziyeqf - Thanks for this PR, quick question below if you can take a look?

Thanks

File: internal/services/securitycenter/advanced_threat_protection_resource.go (hunk 1 above)
Comment (ziyeqf) [reply internal/services/securitycenter/advanced_threat_protection_resource.go]: I'm not sure, taking `nil` as `false` does make sense.

File: internal/services/securitycenter/advanced_threat_protection_resource.go [hunk 2]
```diff
@@ -88,6 +88,35 @@ func resourceAdvancedThreatProtectionCreateUpdate(d *pluginsdk.ResourceData, met
...
+				if respEnabled == d.Get("enabled").(bool) {
+					return resp, "consistent", nil
+				} else {
+					return resp, "diff", nil
+				}
+			}
+			return resp, "error", fmt.Errorf("Properties was nil")
+		},
+		MinTimeout: 1 * time.Minute,
+		Timeout:    time.Until(deadline),
```
Comment (tombuildsstuff) [inline internal/services/securitycenter/advanced_threat_protection_resource.go:113]: presumably if this is eventually consistent we also want to wait for this to be hit a repeated value of times (e.g. 3 times in a row?)

File: internal/services/securitycenter/advanced_threat_protection_resource.go [hunk 3]
```diff
@@ -88,6 +88,36 @@ func resourceAdvancedThreatProtectionCreateUpdate(d *pluginsdk.ResourceData, met
 		return fmt.Errorf("updating Advanced Threat protection for %q: %+v", id.TargetResourceID, err)
 	}
 
+	deadline, _ := ctx.Deadline()
```
Comment (tombuildsstuff) [inline internal/services/securitycenter/advanced_threat_protection_resource.go:91]: we should always ensure the context has a deadline:

```suggestion
	deadline, ok := ctx.Deadline()
	if !ok {
		return fmt.Errorf("internal-error: context had no deadline")
	}
```

**Review by stephybun — APPROVED** [review]
Thanks @ziyeqf LGTM 👍

**Pull request merged**